bot/src/main/java/com/community/tools/service/TaskStatusServiceHooksImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.community.tools.service;
+
+import com.community.tools.dto.UserForTaskStatusDto;
+import com.community.tools.model.TaskNameAndStatus;
+import com.community.tools.model.TaskStatus;
+import com.community.tools.model.stats.UserTask;
+import com.community.tools.repository.stats.UserTaskRepository;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service("taskStatusServiceHooks")
+public class TaskStatusServiceHooksImpl implements TaskStatusService {
+
+  private final UserTaskRepository userTaskRepository;
+  @Value("${tasksForUsers}")
+  private String originalTaskNames;
+
+  public TaskStatusServiceHooksImpl(UserTaskRepository userTaskRepository) {
+    this.userTaskRepository = userTaskRepository;
+  }
+
+  @Override
+  public List<UserForTaskStatusDto> getTaskStatuses(Period period, Integer limit,
+      Comparator<UserForTaskStatusDto> comparator) {
+    LocalDate startDate = LocalDate.ofEpochDay(LocalDate
+        .now()
+        .minus(period).toEpochDay());
+    return userTaskRepository
+        .findUserTasksByLastActivityAfter(startDate)
+        .stream()
+        .collect(Collectors.groupingBy(UserTask::getGitName))
+        .entrySet()
+        .stream()
+        .map(entry -> {
+          List<TaskNameAndStatus> taskNamesAndStatuses = new ArrayList<>();
+          entry.getValue().forEach(
+              task -> taskNamesAndStatuses.add(
+                  new TaskNameAndStatus(task.getTaskName(), task.getPullUrl(),
+                      task.getTaskStatus())));
+
+          final int completedTasksCount = (int) taskNamesAndStatuses.stream()
+              .filter(it -> originalTaskNames.contains(it.getTaskName()))
+              .filter(it -> it.getTaskStatus().equals(
+                  TaskStatus.DONE.getDescription())).count();
+
+          /* This is supposed to throw NoSuchElementException
```

Comment:
I think it would be better to handle NoSuchElementException and write case in logs

Reply:
Good idea, implemented logging